edgedb-cli-derive/src/into_app.rs

```diff
@@ -242,34 +242,37 @@ fn mk_arg(field: &types::Field, case: &Case) -> TokenStream {
             #arg = #arg.#name(#value);
         });
     }
-    match field.parse.kind {
-        ParserKind::TryFromStr => {
-            let ty = &field.ty;
-            let func = if let Some(func) = &field.parse.parser {
-                quote! { #func }
-            } else {
-                quote! { std::str::FromStr::from_str }
-            };
-            modifiers.extend(quote! {
-                #arg = #arg.validator(|v| {
-                    #func(v).map(|_: #ty| ())
+    let validator = syn::Ident::new("validator", Span::call_site());
+    if !field.attrs.options.contains_key(&validator) {
```

Comment:
We must obey `clap` semantics here. Is it what clap does? I'm also not sure why this is needed.

Reply:
Also if we do this check, we should put a flag in `field.attrs` instead of using `contains_key`, as everything else is done.

Reply:
Because we [put a "ParserKind::TryFromStr" for every field that is bool or didn't define a `parse()`](https://github.com/edgedb/edgedb-cli/pull/356/files#diff-ba3f1dea2ab02cd6ab3fa59af7fa5d5e3b4eb9de9c6a8a3756c4f6f7b657d3fbR77-R87), so we'll always append a `validator()` even when `validator()` is explicitly specified. Looks like the second `validator()` will overwrite the first, and that's why `validator(instance_name_opt)` stopped working. I'm open to fixing this in any proper way.

Reply:
Ah, yes. I think we should just move the block that applies arbitrary options (lines 240-244) after applying the parser. So replacing validator just works as expected.

Reply:
Yeah sure, that works too.